lang/attribute/error/src/lib.rs

```diff
@@ -56,6 +59,76 @@ pub fn error(
         false => Some(parse_macro_input!(args as ErrorArgs)),
     };
     let mut error_enum = parse_macro_input!(input as syn::ItemEnum);
-    let error = error_codegen::generate(error_parser::parse(&mut error_enum, args));
+    let error = codegen::error::generate(error_parser::parse(&mut error_enum, args));
     proc_macro::TokenStream::from(error)
 }
+
+/// Generates an [`Error::AnchorError`](../../anchor_lang/error/enum.Error.html) that includes file and line information.
+///
+/// # Example
+/// ```rust,ignore
+/// #[program]
+/// mod errors {
+///     use super::*;
+///     pub fn example(_ctx: Context<Example>) -> Result<()> {
+///         Err(error!(MyError::Hello))
+///     }
+/// }
+///
+/// #[error_code]
+/// pub enum MyError {
+///     #[msg("This is an error message clients will automatically display")]
+///     Hello,
+/// }
+/// ```
+#[proc_macro]
+pub fn error(ts: proc_macro::TokenStream) -> TokenStream {
+    let input = parse_macro_input!(ts as ErrorInput);
+    let error_code = input.error_code;
+    create_error(error_code, true, None)
+}
+
+#[proc_macro]
+pub fn error_without_origin(ts: proc_macro::TokenStream) -> TokenStream {
+    let input = parse_macro_input!(ts as ErrorInput);
+    let error_code = input.error_code;
+    create_error(error_code, false, None)
+}
+
+#[proc_macro]
+pub fn error_with_account_name(ts: proc_macro::TokenStream) -> TokenStream {
```

Comment:
Is this used anywhere? I don't think so. (if not let's remove it)

Reply:
it is used a lot. e.g. in syn/constraints

Reply:
Yea I see it. The diff wasn't showing that file.